PWGHF/D2H/TableProducer/dataCreatorCharmResoReduced.cxx

```diff
@@ -620,6 +620,17 @@ struct HfDataCreatorCharmResoReduced {
     return flagV0; // Placeholder, should return the actual flag based on matching logic
   }
 
+  template <typename PParticles>
+  double computeInvMassGen(PParticles const& particlesMc, int indexRec)
+  {
+    auto particleReso = particlesMc.iteratorAt(indexRec);
+    auto dau1 = particlesMc.iteratorAt(particleReso.daughtersIds().front());
+    auto dau2 = particlesMc.iteratorAt(particleReso.daughtersIds().back());
+    std::array<std::array<float, 3>, 2> pArr = {{{dau1.px(), dau1.py(), dau1.pz()}, {dau2.px(), dau2.py(), dau2.pz()}}};
```

Comment:
you can consider to loop on daughters, to generalise in case of more than 2 daughters